emcc.py

```diff
@@ -387,6 +391,10 @@ def get_clang_flags(user_args):
     # enabled so that it can be added to the attributes in LLVM IR.
     if settings.SUPPORT_LONGJMP == 'wasm':
       flags.append('-mexception-handling')
+    # If this code will be used in a whole-program link then informing clang of
+    # that can help devirtualization.
+    if for_whole_program_link:
+      flags += ['-fwhole-program-vtables']
```

Comment:
Does this need to be in the `if LTO` block.. its possible the non-LTO case might one day benefit too?  

Can you instead put it up on line 377 with the other usage of `for_whole_program_link`?

Reply:
This LLVM feature depends on LTO:

https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-fwhole-program-vtables

It will error otherwise.